services/portal/exportToWorkspace/exportToWorkspaceTasks.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable max-len */
+const exportToWorkspaceProps = require('./exportToWorkspaceProps.js');
+
+const I = actor();
+
+/**
+ * exportToWorkspace Tasks
+ */
+module.exports = {
+  goToWorkspacePage() {
+    I.amOnPage(exportToWorkspaceProps.workspacePath);
+    I.waitForVisible(exportToWorkspaceProps.workspaceIFrameClass, 10);
+  },
+
+  goToExplorerPage() {
+    I.amOnPage(exportToWorkspaceProps.explorerPath);
+    I.waitForVisible(exportToWorkspaceProps.explorerHeaderClass, 10);
+  },
+
+  exportDefaultManifestToWorkspace() {
+    this.goToExplorerPage();
+    I.seeElement(exportToWorkspaceProps.exportToWorkspaceButtonXPath);
+    I.click(exportToWorkspaceProps.exportToWorkspaceButtonXPath);
+  },
+
+  /* Direct to workspace by clicking the `Go to workspace` button in message toaster */
+  jumpToWorkspacePage() {
+    I.seeElement(exportToWorkspaceProps.goToWorkspaceButtonXPath);
+    I.click(exportToWorkspaceProps.goToWorkspaceButtonXPath);
+    I.waitForVisible(exportToWorkspaceProps.workspaceIFrameClass, 10);
+  },
+
+  /* Spawn up a workspace */
+  startWorkspace() {
+    within({ frame: exportToWorkspaceProps.workspaceIFrameClass }, async () => {
+      const start = await I.grabNumberOfVisibleElements(exportToWorkspaceProps.startButtonSelector);
+      if (start === 1) { // see "Start My Server" button
+        I.click(exportToWorkspaceProps.startButtonSelector);
+        const spawn = await I.grabNumberOfVisibleElements(exportToWorkspaceProps.spawnButtonSelector);
+        if (spawn === 1) {
+          I.click(exportToWorkspaceProps.spawnButtonSelector); // spawn with default config
+        }
+      }
+      I.waitForVisible(exportToWorkspaceProps.mainWorkspaceAppSelector, 600); // ipython may be very slow on start up
+    });
+  },
+
+  /* Mount the latest manifest using python codes as defined in 'exportToWorkspaceProps.mountManifestCode' */
+  async mountLatestManifestInJupyterNotebook() {
+    const output = await within({ frame: exportToWorkspaceProps.workspaceIFrameClass }, () => {
+      I.waitForVisible(exportToWorkspaceProps.mainWorkspaceAppSelector, 600);
+      I.click(exportToWorkspaceProps.newDropDownButtonSelector);
+      I.click(exportToWorkspaceProps.newPython3ButtonSelector);
+      I.waitForVisible(exportToWorkspaceProps.pythonNotebookPanelSelector, 60); // wait for new notebook to show up
+      exportToWorkspaceProps.mountManifestCode.forEach((line) => {
+        I.pressKey(line);
+        I.pressKey('Enter');
+      });
+      I.wait(3); // wait for save button becomes clickable
```

Comment:
you do `I.wait` several times, you could use the smartWait function to save a few seconds 😈

Reply:
I was not using `I.wait` here in the first place and hoping smartWait can take care of everything. But sometimes the test will fail at here so I have to add this arbitrary wait to let it work... Not sure why though 🤔

Reply:
i meant this function https://github.com/uc-cdis/gen3-qa/blob/master/utils/apiUtil.js#L170
i see you use `I.wait(3)` several times and it probably doesn't take that long so you could use the smartwait